dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSql.java

```diff
@@ -0,0 +1,292 @@
+        .as(PipelineOptions.class); // FlinkPipelineOptions.class
+    options.setJobName("BeamPlanCreator");
+    Pipeline pipeline = Pipeline.create(options);
+
+    return compilePipeline(sqlStatement, pipeline);
+  }
+
+  /**
+   * compile SQL, and return a {@link Pipeline}.
+   *
+   * <p>This may be changed with following CLI client.
+   */
+  @Experimental
+  public static PCollection<BeamSQLRow> compilePipeline(String sqlStatement, Pipeline basePipeline)
+      throws Exception{
+    PCollection<BeamSQLRow> resultStream = planner.compileBeamPipeline(sqlStatement, basePipeline);
+    return resultStream;
+  }
+
+  //methods for Beam SQL as DSL.
+  /**
+   * Translate a PCollection with type T to {@code PCollection<BeamSQLRow>}
+   * which can be accepted by Beam SQL DSL.
+   */
+  public static <T> PTransform<PCollection<T>, PCollection<BeamSQLRow>> toBeamSqlRow(
+      PTransform<PCollection<T>, PCollection<BeamSQLRow>> recordTransform,
+      BeamSQLRecordType rowType) {
+    return new BeamRowRecordReader<T>(recordTransform, rowType);
+  }
+
+  /**
+   * {@link PTransform} used in {@link BeamSql#toBeamSqlRow(PTransform, BeamSQLRecordType)}
+   * to translate row records.
+   */
+  public static class BeamRowRecordReader<T>
+      extends PTransform<PCollection<T>, PCollection<BeamSQLRow>> {
+    private PTransform<PCollection<T>, PCollection<BeamSQLRow>> recordTransform;
+    private BeamSQLRecordType rowType;
+
+    public BeamRowRecordReader(PTransform<PCollection<T>, PCollection<BeamSQLRow>> recordTransform,
+        BeamSQLRecordType rowType) {
+      this.recordTransform = recordTransform;
+      this.rowType = rowType;
+    }
+
+    @Override
+    public PCollection<BeamSQLRow> expand(PCollection<T> input) {
+      return input.apply(recordTransform).setCoder(new BeamSqlRowCoder(rowType));
+    }
+
+  }
+
+  /**
+   * {@link BeamSql#query(String)} is the core method in Beam SQL DSL.
+   *
+   * <p>It translates a SQL query to a composite {@link PTransform}, which is applied to
+   * {@link PCollectionTuple} and emit a {@link PCollection}. Here {@link PCollectionTuple}
+   * contains the input tables, each is represented as a {@link PCollection}.
+   */
+  public static PTransform<PCollectionTuple, PCollection<BeamSQLRow>> query(String sqlQuery) {
+    return new QueryTransform(sqlQuery);
+
+  }
+
+  /**
+   * {@link BeamSql#simpleQuery(String)} is a light-weight method, with similar functions as
+   * {@link BeamSql#query(String)}, but that it's applied to a single source table.
+   */
+  public static PTransform<PCollection<BeamSQLRow>, PCollection<BeamSQLRow>>
+  simpleQuery(String sqlQuery) throws Exception {
+    return new SimpleQueryTransform(sqlQuery);
+  }
+
+  /**
+   * A {@link PTransform} to translate from one SQL to a Beam composite {@link PTransform}.
+   */
+  public static class QueryTransform extends PTransform<PCollectionTuple, PCollection<BeamSQLRow>> {
```

Comment:
Its fine for now, but you'll see as additional information is added using `@AutoValue` and `@AutoValue.Builder` in the future will help reduce pointless boilerplate and make it easier for other users to make these changes.